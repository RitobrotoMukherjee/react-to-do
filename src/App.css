.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.Todo-main {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 15px auto;
  min-height: 350px;
  width: 65%;
  padding: 30px;
  border: 1px solid rgb(1, 1, 36, 0.85);
  border-radius: 6px;
  box-shadow: 0 3px 15px 15px #cfc6c6;
}

.Add-to-do-form {
  display: flex;
  justify-content: center;
  width: 70%;
  padding: 18px;
  margin: 9px auto;
}

.Form-control {
  width: 70%;
  border: 1px solid #ccc;
  border-radius: 6px;
  padding: 9px;
  margin: 3px auto;
}

.Add-to-do-input {
  padding: 12px;
  border-radius: 21px;
}

.Todo-list {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 70%;
}

.List-item {
  width: 75%;
  padding: 3px;
  margin: 1px auto;
}

.List-item-checkbox {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 12px;
  height: 21px;
  background-color: rgb(15, 228, 15);
  border: 1px solid #ccc;
  border-radius: 60px;
  cursor: pointer;
}

.List-item-checkbox:checked {
  background-color: rgb(244, 115, 98);
}

.Flex-1 {
  flex: 1 1 6px;
  margin: 6px;
}

.Flex-2 {
  flex: 1 1 70%;
}

.Pen-through {
  font-style: italic;
  color: #595959;
  opacity: 0.75;
  text-decoration: line-through;
  pointer-events: none;
}

.Delete-button {
  border: 1px solid rgb(33, 33, 241);
  border-radius: 6px;
  background-color: #fff;
  cursor: pointer;
  color: rgb(33, 33, 241, 0.9);
  transition: all 1s;
}

.Delete-button:hover {
  background-color: rgb(33, 33, 241);
  color: #fff;
  border-radius: 15px;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}
