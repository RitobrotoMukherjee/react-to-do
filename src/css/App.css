.App {
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.MainHeading {
  color: blue;
  font-size: 35px;
  font-weight: 400;
}

.MainHeading::after {
  content: '';
  margin: 3px auto;
  padding: 3px;
  width: 54px;
  border: 1px solid #ccc;
  border-radius: 9px;
  background: blue;
  display: block;
}

.Menu {
  display: flex;
  flex-wrap: wrap;
  flex: 1 1 45%;
  justify-content: center;
  align-items: center;
}

.MenuItems {
  border: 1px solid rgb(133, 4, 79);
  border-radius: 6px;
  padding: 9px;
  margin-right: 45px;
  transition: all 1s;
}

.MenuItems:hover {
  background-color: rgb(133, 4, 79);
}

.Section {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 21px;
  margin: 12px;
  margin-top: 150px;
}

.Todo-main {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 100px auto;
  min-height: 350px;
  width: 65%;
  padding: 30px;
  border: 1px solid rgb(1, 1, 36, 0.85);
  border-radius: 6px;
  box-shadow: 0 3px 15px 15px #cfc6c6;
}

.Add-to-do-form {
  display: flex;
  justify-content: center;
  width: 70%;
  padding: 18px;
  margin: 9px auto;
}

.Form-control {
  width: 70%;
  border: 1px solid #ccc;
  border-radius: 6px;
  padding: 9px;
  margin: 3px auto;
}

.Add-to-do-input {
  padding: 12px;
  border-radius: 21px;
}

.Todo-list {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 70%;
}

.List-item {
  width: 75%;
  padding: 3px;
  margin: 1px auto;
}

.List-item-checkbox {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  width: 12px;
  height: 21px;
  background-color: rgb(15, 228, 15);
  border: 1px solid #ccc;
  border-radius: 60px;
  cursor: pointer;
}

.List-item-checkbox:checked {
  background-color: rgb(244, 115, 98);
}

.Flex-1 {
  flex: 1 1 6px;
  margin: 6px;
}

.Flex-2 {
  flex: 1 1 70%;
}

.Pen-through {
  font-style: italic;
  color: #595959;
  opacity: 0.75;
  text-decoration: line-through;
  pointer-events: none;
}

.Delete-button {
  border: 1px solid rgb(33, 33, 241);
  border-radius: 6px;
  background-color: #fff;
  cursor: pointer;
  color: rgb(33, 33, 241, 0.9);
  transition: all 1s;
}

.Delete-button:hover {
  background-color: rgb(33, 33, 241);
  color: #fff;
  border-radius: 15px;
}

.ParagraphContainer {
  display: flex;
  flex-direction: column;
  text-align: justify;
}
