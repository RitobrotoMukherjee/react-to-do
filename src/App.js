import './App.css';
import TodoList from './components/TodoList';

function App() {
  return (
    <section className="App">
      <TodoList />
    </section>
  );
}

export default App;
