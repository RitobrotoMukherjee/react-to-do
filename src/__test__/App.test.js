import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import App from '../App';

describe('Test components', () => {
  test('Test App Component Snapshot', () => {
    // Arrange, Act
    const TREE = renderer.create(<App />).toJSON();
    // Assert
    expect(TREE).toMatchSnapshot();
  });

  test('Test App component Rendering', () => {
    // Arrage
    render(<App />);

    // Act
    const FORMS = screen.getAllByPlaceholderText('Add A To Do');

    // Assert
    expect(screen.getByText('To Do App')).toBeInTheDocument();
    expect(FORMS.length).toBe(1);
  });
});
