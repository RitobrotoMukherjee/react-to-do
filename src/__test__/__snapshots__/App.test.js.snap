// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test components Test App Component Snapshot 1`] = `
<section
  className="App"
>
  <div
    className="Todo-main"
  >
    <header
      style={
        Object {
          "lineHeight": "1.5em",
          "padding": "20px 0",
        }
      }
    >
      <h1
        style={
          Object {
            "color": "blue",
            "fontSize": "35px",
            "fontWeight": "400",
            "lineHeight": "1em",
            "marginBottom": "15px",
            "textAlign": "center",
          }
        }
      >
        To Do App
      </h1>
    </header>
    <form
      className="Add-to-do-form"
      onSubmit={[Function]}
    >
      <input
        className="Form-control Add-to-do-input"
        name="title"
        onChange={[Function]}
        placeholder="Add A To Do"
        value=""
      />
    </form>
    <ul
      className="Todo-list"
    />
  </div>
</section>
`;
